great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -1,31 +1,95 @@
 from __future__ import annotations
 
+import functools
 import logging
-from typing import TYPE_CHECKING, Final, Literal, Optional, Type, Union
+import urllib.parse
+import warnings
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Final,
+    Iterable,
+    Literal,
+    Optional,
+    Type,
+    Union,
+)
 
-from great_expectations._docs_decorators import public_api
+from great_expectations._docs_decorators import deprecated_method_or_class, public_api
 from great_expectations.compatibility import pydantic
 from great_expectations.compatibility.pydantic import AnyUrl, errors
 from great_expectations.compatibility.snowflake import URL
 from great_expectations.compatibility.sqlalchemy import sqlalchemy as sa
 from great_expectations.compatibility.typing_extensions import override
+from great_expectations.datasource.fluent import GxContextWarning, GxDatasourceWarning
 from great_expectations.datasource.fluent.config_str import (
     ConfigStr,
+    ConfigUri,
     _check_config_substitutions_needed,
 )
 from great_expectations.datasource.fluent.sql_datasource import (
     FluentBaseModel,
     SQLDatasource,
     SQLDatasourceError,
+    TableAsset,
 )
 
 if TYPE_CHECKING:
     from great_expectations.compatibility import sqlalchemy
     from great_expectations.compatibility.pydantic.networks import Parts
+    from great_expectations.datasource.fluent.interfaces import (
+        BatchMetadata,
+        SortersDefinition,
+    )
     from great_expectations.execution_engine import SqlAlchemyExecutionEngine
 
 LOGGER: Final[logging.Logger] = logging.getLogger(__name__)
 
+REQUIRED_QUERY_PARAMS: Final[Iterable[str]] = {  # errors will be thrown if any of these are missing
+    # TODO: require warehouse and role
+    # "warehouse",
+    # "role",
+}
+
+MISSING: Final = object()  # sentinel value to indicate missing values
+
+
+@functools.lru_cache(maxsize=4)
+def _extract_path_sections(url_path: str) -> dict[str, str]:
```

Comment:
Can we make this name more descriptive, since it's doing something more specific? Like `_extract_database_and_schema_path_sections`.

Reply:
I'll do this as an immediate followup.